src/Microsoft.IdentityModel.Tokens/InMemoryCryptoProviderCache.cs

```diff
@@ -84,23 +84,42 @@ private string GetCacheKeyPrivate(SecurityKey securityKey, string algorithm, str
             return string.Format(CultureInfo.InvariantCulture,
                                  "{0}-{1}-{2}-{3}",
                                  securityKey.GetType(),
-                                 string.IsNullOrEmpty(securityKey.KeyId) ? securityKey.InternalId : securityKey.KeyId,
+                                 securityKey.InternalId,
                                  algorithm,
                                  typeofProvider);
         }
 
+        /// <summary>
+        /// For some security key types, in some runtimes, it's not possible to extract public key material and create <see cref="SecurityKey.InternalId"/>.
```

Comment:
minor grammar nit: create **a**

Reply:
Replaced with "create an", as "create an InternalId" is what will be shown on MSFT docs website.